### PR TITLE
Misc property pages fixes

### PR DESCRIPTION
**Pull request by RaulPerez1**
Second attempt at fixing #813. Implementing a provider for OutputTypeEx that will transform the value of OutputType to the equivalent enum value (uint) for VSLangProj110.prjOutputTypeEx. It's still not 100% as the descriptor for the values will still return a string instead of an int but the string can be safely casted now to uint which is what the code was mostly doing before except for one instance where it expected an int directly. 

Also did some minor cleanup on the rules file and added the .NetCore to the check of IsTargetingDotNetFramework45OrAbove which is used in conjunction with the output type for determining if prefer 32 bit should be enabled or not.

### CODE REVIEW

**Comment by 333fred** [comment]
@dotnet-bot retest Windows Release please

Another instance of https://github.com/dotnet/roslyn-project-system/issues/837.

File: src/Microsoft.VisualStudio.AppDesigner/PropPages/PropPageUserControlBase.vb [hunk 1]
```diff
@@ -812,7 +812,7 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
                 If ProjectProperties IsNot Nothing Then
                     Return ProjectProperties.OutputType
                 Else
-                    Dim obj As Object = TryGetNonCommonPropertyValue(GetPropertyDescriptor("OutputType"))
+                    Dim obj As Object = TryGetNonCommonPropertyValue(GetPropertyDescriptor("OutputTypeEx"))
```
Comment (333fred) [inline src/Microsoft.VisualStudio.AppDesigner/PropPages/PropPageUserControlBase.vb]: As @davkean has mentioned before, I'm concerned about functional changes to the editor. Is this an actual bug in the implementation, or will this change behavior for non-core apps?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/OutputTypeExValueProvider.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                        value = "3";
+                        break;
+                    case "appcontainerexe":
+                        // prjOutputTypeEx_AppContainerExe
+                        value = "4";
+                        break;
+                }
+            }
+
+            return await Task.FromResult<string>(value).ConfigureAwait(false);
```
Comment (333fred) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/OutputTypeExValueProvider.cs]: Just return the value.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralBrowseObject.xaml [hunk 3]
```diff
@@ -65,8 +65,8 @@
   </StringProperty>
   <StringProperty Name="DefineConstants" DisplayName="Define Constants" Visible="True"/>
   <EnumProperty Name="PlatformTarget" DisplayName="Platform Target" Visible="True"/>
-  <StringProperty Name="Prefer32Bit" DisplayName="Prefer 32Bit" Visible="True"/>
-  <StringProperty Name="AllowUnsafeBlocks"  Default="False"  DisplayName="Allow unsafe code" Visible="True"/>
+  <BoolProperty Name="Prefer32Bit" DisplayName="Prefer 32Bit" Visible="True"/>
```
Comment (333fred) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralBrowseObject.xaml]: What are these changes? They seem unrelated?

File: src/Microsoft.VisualStudio.AppDesigner/PropPages/PropPageUserControlBase.vb (hunk 1 above)
Comment (RaulPerez1) [inline src/Microsoft.VisualStudio.AppDesigner/PropPages/PropPageUserControlBase.vb]: According to the comment that code path is specific to CPS. I'm not sure if this ever worked for CPS projects but what it was doing before was grabbing OutputType while all other places (including the dropdown) are grabbing OutputTypeEx so that's probably a bug in the initial implementation. The summary of these 2 should be - OutputType will return the string for the property (i.e. "Exe") while OutputTypeEx will return the transformation of that into a VSLangProj110.prjOutputTypeEx value.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralBrowseObject.xaml (hunk 3 above)
Comment (RaulPerez1) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralBrowseObject.xaml]: Death by a thousand cuts - there's a bunch of things property page related that are not quite right. In this specific case those values should have been Boolean not strings. That fixes the dropdown correctly in the property grid. This PR has 4 changes, they're split up into different commits as each one addresses a different thing.

**Comment by RaulPerez1** [comment]
Added fixes for #415, #823 and #977 I can split it up in multiple PR's later but some of the changes depend on previous fixes.

File: src/Microsoft.VisualStudio.Editors/Common/Utils.vb [hunk 4]
```diff
@@ -1566,6 +1566,56 @@ Namespace Microsoft.VisualStudio.Editors.Common
         ''' <remarks></remarks>
         Friend Function IsTargetingDotNetFramework45OrAbove(hierarchy As IVsHierarchy) As Boolean
 
+            Dim frameworkName As FrameworkName = Nothing
+            If TryGetTargetFrameworkMoniker(hierarchy, frameworkName) Then
+                If String.Compare(frameworkName.Identifier, ".NETFramework", StringComparison.OrdinalIgnoreCase) = 0 AndAlso
+                (frameworkName.Version.Major > 4 OrElse (frameworkName.Version.Major = 4 AndAlso frameworkName.Version.Minor >= 5)) Then
+                    ' Targeting .NET framework 4.5 or higher
+                    Return True
+                ElseIf IsTargetingDotNetCore(frameworkName.Identifier) Then
```
Comment (333fred) [inline src/Microsoft.VisualStudio.Editors/Common/Utils.vb]: I'm rather concerned by this. Is this a public function that other people consume, or just internal? Because if it's just internal, we shouldn't return true for this case: it's *not* targeting .netframework 4.5 or above, it's targeting .netcore 1.0. And even if it is public, we should fix the thing calling this, not return true when the condition is actually false.

File: src/Microsoft.VisualStudio.Editors/PropPages/ApplicationPropPage.vb [hunk 5]
```diff
@@ -676,6 +676,8 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
 
             'Populate the target framework combobox
             PopulateTargetFrameworkComboBox(Me.TargetFramework)
+
+            Me.AssemblyInfoButton.Visible = Not Utils.IsTargetingDotNetCore(Me.ProjectHierarchy)
```
Comment (davkean) [inline src/Microsoft.VisualStudio.Editors/PropPages/ApplicationPropPage.vb]: Hardcoding TFMs is definitely not the right approach. We're also enabling this for *desktop* projects that are based in CPS.

File: src/Microsoft.VisualStudio.Editors/Common/Utils.vb (hunk 4 above)
Comment (RaulPerez1) [inline src/Microsoft.VisualStudio.Editors/Common/Utils.vb]: The main problem is that when the pages were made it was presuming that there wouldn't be anything other than .Net framework for those.

With that in mind this is internal for the utils and and only used for the Prefer32Bit check. The other option here is to change the check in the property page to call both functions.

File: src/Microsoft.VisualStudio.Editors/PropPages/ApplicationPropPage.vb (hunk 5 above)
Comment (RaulPerez1) [inline src/Microsoft.VisualStudio.Editors/PropPages/ApplicationPropPage.vb]: I agree and looking at the code I think I found one other bug - The package page will show up for all projects right now so first we need to add a mechanism to check for this. With that in mind, the property pages have no knowledge of CPS right now so they wouldn't be able to do capabilities. I think we need to introduce a property (that can be backed by a capability) which then the property pages can consume.

Comment (srivatsn) [inline src/Microsoft.VisualStudio.Editors/PropPages/ApplicationPropPage.vb]: Capabilties are not a CPS-only concept. There's a hierarchy property you can query to get the project node's capabilities I think.

Comment (RaulPerez1) [inline src/Microsoft.VisualStudio.Editors/PropPages/ApplicationPropPage.vb]: Made the change and opened #1001 to track the other bug. Once that's implemented the button will be disabled.

File: src/Microsoft.VisualStudio.Editors/Common/Utils.vb [hunk 6]
```diff
@@ -1644,6 +1684,25 @@ Namespace Microsoft.VisualStudio.Editors.Common
             Return False
         End Function
 
+        ''' <summary>
+        ''' Gets a value indicating whether the project associated with the given hierarchy contains the specified capability.
+        ''' </summary>
+        ''' <param name="hierarchy">Hierarchy object.</param>
+        ''' <returns>Value indicating whether the specified capability is present</returns>
+        Friend Function ContainsCapability(hierarchy As IVsHierarchy, capability As String) As Boolean
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.Editors/Common/Utils.vb]: #1023 is using a shell extension method for this. So this can be removed.

File: src/Microsoft.VisualStudio.Editors/PropPages/AdvBuildSettingsPropPage.Designer.vb [hunk 7]
```diff
@@ -92,7 +92,7 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
             resources.ApplyResources(Me.cboDebugInfo, "cboDebugInfo")
             Me.cboDebugInfo.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList
             Me.cboDebugInfo.FormattingEnabled = True
-            Me.cboDebugInfo.Items.AddRange(New Object() {resources.GetString("cboDebugInfo.Items"), resources.GetString("cboDebugInfo.Items1"), resources.GetString("cboDebugInfo.Items2")})
+            Me.cboDebugInfo.Items.AddRange(New Object() {resources.GetString("cboDebugInfo.Items"), resources.GetString("cboDebugInfo.Items1"), resources.GetString("cboDebugInfo.Items2"), resources.GetString("cboDebugInfo.Items3"), resources.GetString("cboDebugInfo.Items4")})
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.Editors/PropPages/AdvBuildSettingsPropPage.Designer.vb]: What does this do? What effect will this have on desktop projects?

File: src/Microsoft.VisualStudio.Editors/PropPages/ApplicationPropPage.vb [hunk 8]
```diff
@@ -676,6 +676,8 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
 
             'Populate the target framework combobox
             PopulateTargetFrameworkComboBox(Me.TargetFramework)
+
+            Me.AssemblyInfoButton.Visible = Not Utils.ContainsCapability(Me.ProjectHierarchy, "PackagePage")
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.Editors/PropPages/ApplicationPropPage.vb]: Not needed due to #1023

File: src/Microsoft.VisualStudio.Editors/PropPages/AdvBuildSettingsPropPage.Designer.vb (hunk 7 above)
Comment (RaulPerez1) [inline src/Microsoft.VisualStudio.Editors/PropPages/AdvBuildSettingsPropPage.Designer.vb]: This is the fix for #415 - It will make it so that Portable and Compact show up in the dialog and according to the discussion in there those should show up for all projects not just .Net Core.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/PropertyPage.cs [hunk 9]
```diff
@@ -23,6 +23,7 @@ public abstract partial class PropertyPage : UserControl,
         private bool _useJoinableTaskFactory = true;
         private IVsDebugger _debugger;
         private uint _debuggerCookie;
+        private bool isActivated = false;
```
Comment (333fred) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/PropertyPage.cs]: Will this always be modified on the same thread?

Comment (RaulPerez1) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/PropertyPage.cs]: Init happens on the main thread
